### PR TITLE
HHH-14033 - SQL script parsing problem with multi-line comments

### PR DESCRIPTION
**Pull request by sebersole**
- Better handling of multi-line comments
- Restructured some internal classes to consolidate packages
- Added "system"-style SchemaToolingLogging

https://hibernate.atlassian.net/browse/HHH-14033

### CODE REVIEW

**Comment by sebersole** [comment]
Consider specifically whether y'all think it is appropriate to switch the exception type for the missing `;` error

**Comment by dreab8** [comment]
Hi Steve,  the specific message was introduce to fix https://github.com/quarkusio/quarkus/issues/3817, but I do not think that changing the exception type is a problem.

File: hibernate-core/hibernate-core.gradle [hunk 1]
```diff
@@ -203,6 +203,9 @@ xjc {
     }
 }
 
+generateGrammarSource {
+    arguments += "-traceParser"
```
Comment (Sanne) [inline hibernate-core/hibernate-core.gradle:207]: Is this going to affect the regular HQL parser as well?

Specifically I'm wondering about performance impact, if this is to help during development maybe it shoduld be commented out when done with debugging?

Comment (sebersole) [reply hibernate-core/hibernate-core.gradle:207]: The impact - if you look in the generated parsers, you will now see Antlr has added traceIn/traceOut calls at the start and end of each rule method.  I use that as hooks to debug "parse tree info".  This is extremely useful information for debugging parse problems; unless we can demonstrate that it is a performance problem I'd prefer to leave these as-is.

Comment (sebersole) [reply hibernate-core/hibernate-core.gradle:207]: Its a good point though in that the default impl Antlr injects is to use sysout.  For HQL and for SQL script parsing I've overridden these to use logging.  I'll add the same override for the entity-graph parser and really the impact overall should be minimal

**Comment by sebersole** [comment]
Right - I figured.  Which is why I kept the message unchanged

**Comment by sebersole** [comment]
Going to push an additional commit.  If y'all agree I'll just puch this to master

**Comment by dreab8** [comment]
@sebersole it is fine to me

**Comment by Sanne** [comment]
> Going to push an additional commit. If y'all agree I'll just push this to master

sure!

**Pull request merged**